docs/pages/versions/unversioned/sdk/media-library-next.mdx

```diff
@@ -0,0 +1,232 @@
+<SnackInline label="Add a new asset from the web" dependencies={['expo-file-system', 'expo-image', 'expo-media-library']}>
+
+```tsx
+import { View, Text } from 'react-native';
+import { Image } from 'expo-image';
+import { useEffect, useState } from 'react';
+import { File, Paths } from 'expo-file-system';
+import { Asset, requestPermissionsAsync } from 'expo-media-library/next';
+
+export default function SaveToMediaLibraryExample() {
+  const [asset, setAsset] = useState<Asset | null>(null);
+
+  const downloadFile = async () => {
+    const url = 'https://picsum.photos/200/300';
+    const destinationFile = new File(Paths.cache, 'test_image.jpg');
+    if (destinationFile.exists) {
+      return destinationFile;
+    } else {
+      return File.downloadFileAsync(url, destinationFile);
+    }
+  };
+
+  useEffect(() => {
+    async function downloadAndSaveAsset() {
+      const file = await downloadFile();
+
+      const { status } = await requestPermissionsAsync();
+      if (status !== 'granted') return;
+
+      const asset = await Asset.create(file.uri);
+      setAsset(asset);
+    }
+
+    downloadAndSaveAsset();
+  }, []);
+
+  return (
+    <View>
+      {asset ? (
+        <>
+          <Text>{asset.id}</Text>
+          <Image source={{ uri: asset.id }} style={{ width: 200, height: 300 }} />
+        </>
+      ) : (
+        <Text>Downloading and creating asset...</Text>
+      )}
+    </View>
+  );
+}
+```
+
+</SnackInline>
+</Collapsible>
+
+<Collapsible summary="Retrieve asset properties">
+<SnackInline label="Retrieve asset properties" dependencies={["expo-media-library"]}>
+
+```tsx
+import { View, Text } from 'react-native';
+import { useEffect, useState } from 'react';
+import { AssetField, MediaType, Query, requestPermissionsAsync } from 'expo-media-library/next';
+
+export default function RetrievingAssetPropertiesExample() {
+  const [assetInfo, setAssetInfo] = useState<{
+    id: string;
+    filename: string;
+    mediaType: string;
+    width: number;
+    height: number;
+    creationTime: number | null;
+    modificationTime: number | null;
+  } | null>(null);
+
+  useEffect(() => {
+    const querySomeAsset = async () => {
+      await requestPermissionsAsync();
+      const [asset] = await new Query().limit(1).eq(AssetField.MEDIA_TYPE, MediaType.IMAGE).exe();
+      if (asset) {
+        const filename = await asset.getFilename();
+        const mediaType = (await asset.getMediaType()).toString();
+        const width = await asset.getWidth();
+        const height = await asset.getHeight();
+        const creationTime = await asset.getCreationTime();
+        const modificationTime = await asset.getModificationTime();
+        setAssetInfo({
+          id: asset.id,
+          filename,
+          mediaType,
+          width,
+          height,
+          creationTime,
+          modificationTime,
+        });
+      } else {
+        console.log('No assets found in the media library.');
+      }
+    };
```

Comment:
I think the examples are good, but all of them follow a slightly different pattern of line breaks before and after `if` statements, function calls etc. Let's keep them consistent.